be/src/exec/pipeline/scan/olap_meta_chunk_source.cpp

```diff
@@ -0,0 +1,45 @@
+// This file is licensed under the Elastic License 2.0. Copyright 2021-present, StarRocks Inc.
+
+#include "exec/pipeline/scan/olap_meta_chunk_source.h"
+
+#include "exec/pipeline/scan/olap_meta_scan_operator.h"
+#include "exec/workgroup/work_group.h"
+
+namespace starrocks::pipeline {
+
+OlapMetaChunkSource::OlapMetaChunkSource(int32_t scan_operator_id, RuntimeProfile* runtime_profile, MorselPtr&& morsel,
+                                         vectorized::OlapMetaScanNode* scan_node, OlapMetaScanContextPtr scan_ctx)
+        : ChunkSource(scan_operator_id, runtime_profile, std::move(morsel), scan_ctx->get_chunk_buffer()),
+          _scan_node(scan_node),
+          _scan_ctx(scan_ctx) {}
+
+OlapMetaChunkSource::~OlapMetaChunkSource() {}
+
+Status OlapMetaChunkSource::prepare(RuntimeState* state) {
+    RETURN_IF_ERROR(ChunkSource::prepare(state));
+    // use tablet id in morsel to get olap meta scanner
+    auto scan_morsel = dynamic_cast<ScanMorsel*>(_morsel.get());
```

Comment:
what if cast failed?

Reply:
> what if cast failed?

it won't happen, I add a DCHECK under this line…